blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/AbstractElasticIO.java

```diff
@@ -51,101 +46,57 @@
         return search(tenant, Arrays.asList(query));
     }
 
-
     public List<SearchResult> search(String tenant, List<String> queries) throws Exception {
-        String[] indexes = getIndexesToSearch();
-
-        return searchESByIndexes(tenant, queries, indexes);
+        return searchUsingRestApi(tenant, queries);
     }
 
-    private List<SearchResult> searchESByIndexes(String tenant, List<String> queries, String[] indexes) {
-        List<SearchResult> results = new ArrayList<SearchResult>();
+    private List<SearchResult> searchUsingRestApi(String tenant, List<String> queries) {
+        List<SearchResult> results = new ArrayList<>();
         Timer.Context multiSearchCtx = searchTimer.time();
-        SearchResponse response;
+        int hitsCount = 0;
+
         try {
             queryBatchHistogram.update(queries.size());
-            BoolQueryBuilder bqb = boolQuery();
-            QueryBuilder qb;
-
-            for (String query : queries) {
-                GlobPattern pattern = new GlobPattern(query);
-                if (!pattern.hasWildcard()) {
-                    qb = termQuery(ESFieldLabel.metric_name.name(), query);
-                } else {
-                    qb = regexpQuery(ESFieldLabel.metric_name.name(), pattern.compiled().toString());
-                }
-
-                bqb.should(boolQuery()
-                                .must(termQuery(ESFieldLabel.tenantId.toString(), tenant))
-                                .must(qb)
-                );
-            }
-
-            response = client.prepareSearch(indexes)
-                    .setRouting(tenant)
-                    .setSize(MAX_RESULT_LIMIT)
```

Comment:
What does this limit do, and does it need to be preserved in the rest client?

Reply:
This sets the limit of the returned result size.

Reply:
That's Elasticsearch max_result_window limit. https://www.elastic.co/guide/en/elasticsearch/reference/2.4/index-modules.html.

Reply:
my question is that the new code seems to have changed the max from 100,000 to 10,000.  Is that a true statement?  if so, why?

Reply:
As discussed, this limit is added in the newer version of Elasticsearch. Here is the exception that elasticsearch throws:
`QueryPhaseExecutionException[Result window is too large, from + size must be less than or equal to: [10000] but was [100000]. See the scroll api for a more efficient way to request large data sets. This limit can be set by changing the [index.max_result_window] index level parameter.]`